executor/executor.go

```diff
@@ -234,8 +236,23 @@ func (e *IndexRangeExec) Next() (*Row, error) {
 	}
 }
 
+// indexCompare compares multi column index.
+// The length of boundVals may be less than idxKey.
+func indexCompare(idxKey []interface{}, boundVals []interface{}) (int, error) {
```

Comment:
Actually, the `indexSingleCompare` is not the whole index, but a single column in an index.